static/server_configuration.sh

```diff
@@ -37,9 +37,14 @@ choice=$(whiptail --title "Server configurations" --checklist "Choose what you w
 "Security" "(Add extra security based on this http://goo.gl/gEJHi7)" OFF \
 "Static IP" "(Set static IP in Ubuntu with netplan.io)" OFF \
 "Disk Check" "(Check for S.M.A.R.T errors on your disks every week on Mondays)" OFF \
+"Bitwarden Registration" "(Enable or disable public user registration for Bitwarden)" OFF \
 "Automatic updates" "(Automatically update your server every week on Sundays)" OFF 3>&1 1>&2 2>&3)
 
 case "$choice" in
+    *"Bitwarden Registration"*)
+        clear
+        run_script APP bitwarden-registration
+    ;;&
```

Comment:
We should think about the best order executing those scripts: I am not sure if when all options get chosen in server_configuration, the Bitwarden Registration gets executed first and the Disk Check second, and so on...

Reply:
Please test it and make proper changes.

And yes, radiolist would fit better. :+1: Was quick and dirty yesterday. :)

Reply:
So my idea is to use this order here:
```
"Activate TLS" "(Enable HTTPS with Let's Encrypt)" ON \
"Security" "(Add extra security based on this http://goo.gl/gEJHi7)" OFF \
"Static IP" "(Set static IP in Ubuntu with netplan.io)" OFF \
"Automatic updates" "(Automatically update your server every week on Sundays)" OFF \
"Disk Check" "(Check for S.M.A.R.T errors on your disks every week on Mondays)" OFF \
"Bitwarden Registration" "(Enable or disable public user registration for Bitwarden)" OFF 3>&1 1>&2 2>&3)
```
and execute it in the same order (apart from activate-tls, which should probably get executed always at the end):

```
case "$choice" in
    *"Security"*)
        clear
        run_script ADDONS security
    ;;&
    *"Static IP"*)
        clear
        run_script NETWORK static_ip
    ;;&
    *"Automatic updates"*)
        clear
        run_script ADDONS automatic_updates
    ;;&
    *"Disk Check"*)
        clear
        run_script DISK smartctl
    ;;&
    *"Bitwarden Registration"*)
        clear
        run_script APP bitwarden-registration
    ;;&
    *"Activate TLS"*)
        clear
msg_box "The following script will install a trusted
TLS certificate through Let's Encrypt.
It's recommended to use TLS (https) together with Nextcloud.
Please open port 80 and 443 to this servers IP before you continue.
More information can be found here:
https://www.techandme.se/open-port-80-443/"

        if [[ "yes" == $(ask_yes_or_no "Do you want to install TLS?") ]]
        then
            if [ -f $SCRIPTS/activate-tls.sh ]
            then
                bash $SCRIPTS/activate-tls.sh
            else
                download_script LETS_ENC activate-tls
                bash $SCRIPTS/activate-tls.sh
            fi
        else
            echo
            print_text_in_color "$ICyan" "OK, but if you want to run it later, just type: sudo bash $SCRIPTS/activate-tls.sh"
            any_key "Press any key to continue..."
        fi

        # Just make sure they are gone
        rm -f "$SCRIPTS/test-new-config.sh"
        rm -f "$SCRIPTS/activate-tls.sh"
        clear
    ;;&
    *)
    ;;
esac
```

Reply:
I've moved that proposal to https://github.com/nextcloud/vm/pull/1351